network/dag/state.go

```diff
@@ -107,6 +105,21 @@ func (s *state) RegisterObserver(observer Observer, transactional bool) {
 
 }
 
+func (s *state) treeObserver(ctx context.Context, transaction Transaction, payload []byte) error {
```

Comment:
should the be tested?

Reply:
added tests for non-happy flows. Happy flow is tested all over the place